features/207-sdk-api-consistency/record-api.md

```diff
@@ -240,25 +353,64 @@ saveNonAtomically(records: Record[]): Promise<NonAtomicSaveResult[]>;
 
 ##### New
 
+- Deleted record is an `SKYRecord` Object that contain `recordID`, `recordType` and `deleted`,
+`deleted` is `true` and the other part of record is empty.
+
 ```objc
 - (void)deleteRecordWithType:(NSString *)recordType
                     recordID:(NSString *)recordID
-           completionHandler:(void (^_Nullable)(NSString *_Nullable recordID,
-                                                NSError *_Nullable error))completionHandler;
+                  completion:(void (^_Nullable)(NSString *_Nullable recordID,
+                                                NSError *_Nullable error))completion;
 
 - (void)deleteRecordsWithType:(NSString *)recordType
                     recordIDs:(NSArray<NSString *> *)recordIDs
-           completionHandler:(void (^_Nullable)(NSArray *_Nullable deletedRecordIDs,
-                                                NSError *_Nullable error))completionHandler;
+                   completion:(void (^_Nullable)(NSArray *_Nullable deletedRecordIDs,
+                                                 NSError *_Nullable error))completion;
+
+- (void)deleteRecord:(SKYRecord *)record
+          completion:(void (^_Nullable)(SKYRecord *_Nullable deleteRecord,
+                                        NSError *_Nullable error))completion;
+
+- (void)deleteRecords:(NSArray<SKYRecord*> *)records
+           completion:(void (^_Nullable)(NSArray *_Nullable deleteRecords,
+                                         NSError *_Nullable error))completion;
+
+// Generic RecordResult object for fetch, non-atomic save and delete
+@interface SKYRecordResult<ObjectType> : NSObject
 
+@property (nonatomic, readonly) ObjectType value;
+@property (nonatomic, readonly) NSError* error;
+
+@end
+// results contain deleted record id
 - (void)deleteRecordsWithTypeNonAtomically:(NSString *)recordType
```

Comment:
See if you want to put NonAtomically to the middle so that `deleteRecordsNonAtomicallyWithType:`

Reply:
Updated, thanks!